### PR TITLE
Run setfiles after config.sh (bsc#1210604)

### PR DESCRIPTION
**Pull request by Vogtinator**
config.sh can perform changes which need to be labelled correctly.

### CODE REVIEW

**Review by schaefi — APPROVED** [review]
Looks good to me :+1: Thanks

**Comment by Vogtinator** [comment]
This is not quite enough because kiwi itself changes files later on, like `/etc/sysconfig/bootloader` and `/etc/sysconfig/proxy`. Those are then unlabelled in the image.

So it appears like the relabelling needs to happen even later.

**Comment by Vogtinator** [comment]
`/etc/fstab` as well, particularly in the context of using an `fstab.script`

**Review by davidcassany — COMMENTED** [review]
> This is not quite enough because kiwi itself changes files later on, like /etc/sysconfig/bootloader and /etc/sysconfig/proxy. Those are then unlabelled in the image.

I see, this is a problem, this might be tricky, tracking modified files after prepare step is not an easy task. Probably we could consider recalling again a relabel within `ImageBuilder.create()` for some specific types. I dislike the idea but I doubt we can easily find a common stage to label those as those are likely coupled to image type and its attributes.

Regarding the fstab.script, images.sh and other hooks having potential to modify the root-tree on later stages I'd say we should essentially instruct to relabel files within the script itself.

File: kiwi/tasks/system_build.py [hunk 1]
```diff
@@ -283,6 +282,9 @@ def process(self):
         # call config.sh script if present
         setup.call_config_script()
 
+        # if configured, assign SELinux labels
+        setup.setup_selinux_file_contexts()
+
         # handle uninstall package requests, gracefully uninstall
         # with dependency cleanup
         system.pinch_system(force=False)
```
[the remark this answers is not in the document]
Comment (Vogtinator) [reply kiwi/tasks/system_build.py:290]: Yeah, I'm not sure either. It's also technically possible that a system is created in such a way that it has SELinux enabled but does not include the necessary tools for relabelling within the image at this point anymore.

**Comment by Vogtinator** [comment]
> Currently I'm trying to do a relabel from within fstab.script which is kind of a hack, but does also work with older kiwi and matches your suggestion as well...

Works! Given that this hack isn't that ugly and works both with current kiwi and fixed kiwi in the future, I'll use that for now.

I'd still like to fix this in kiwi, but I'm not entirely sure how. Maybe move the relabel to a place after the `chroot image-root fstab.script`?

**Comment by davidcassany** [comment]
> I'd still like to fix this in kiwi, but I'm not entirely sure how. Maybe move the relabel to a place after the `chroot image-root fstab.script`?

I don't think moving is the right way. IMHO the prepare step should already end with the labels set and ready. Then probably recall it in build step for disk images. I'd say the disk.sh is better for that matter this is the last hook in disks I think. Probably after calling disk.sh we could relabel again on disks and reuse the same chroot env created for the disk.sh.

**Comment by Conan-Kudo** [comment]
If we can get labeling to happen after everything has been synchronized to the "disk", that may solve the issues we're having with Fedora Asahi and some CentOS images too.

**Comment by davide125** [comment]
I can confirm doing the relabeling in `fstab.script`  works, see https://pagure.io/fedora-asahi/kiwi-descriptions/c/9b185dae8ac67d1bd55a94f9f65a0029624f7621?branch=rawhide for the workaround I put in place in the Fedora Asahi Remix to replace the autorelabel on first boot we were doing before.

**Comment by schaefi** [comment]
> > I'd still like to fix this in kiwi, but I'm not entirely sure how. Maybe move the relabel to a place after the `chroot image-root fstab.script`?
> 
> I don't think moving is the right way. IMHO the prepare step should already end with the labels set and ready. Then probably recall it in build step for disk images. I'd say the disk.sh is better for that matter this is the last hook in disks I think. Probably after calling disk.sh we could relabel again on disks and reuse the same chroot env created for the disk.sh.

I like this idea. So let's move ```setup_selinux_file_contexts``` to the very end in ```./tasks/system_build.py``` and ```./tasks/system_prepare.py```. Next modify ```_call_script``` and ```_call_script_no_chroot``` from ```./system/setup.py``` to recall setfiles

That should bring us on the safe side

@davidcassany / @Vogtinator would one of you be willing to code up a pull request for this part ? actually the first part is almost done. I'm asking because this PR is in draft mode for quite some time and I'm not sure if it was forgotten due to the fstab.script workaround

Thanks

**Comment by schaefi** [comment]
@Vogtinator I hope it's ok as I didn't saw progress on this one, I took the changes you did and added the changes I think we are still missing and combine that in a new PR #2406

**Pull request closed without merging**